src/Directory.Build.targets

```diff
@@ -8,4 +8,31 @@
   <Import Project="packaging.stubs.targets" Condition="'$(MSBuildProjectExtension)' != '.pkgproj'" />
 
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory).., Directory.Build.targets))\Directory.Build.targets" />
+
+  <!--
+    Package build can be skipped by invoking PackIfServiced instead of Pack targets.
+
+    ProjectServicingConfiguration is used to determine if packaging should be skipped. See more in eng/Versions.props.
+  -->
+  <Target Name="GetCurrentProjectServicingConfiguration">
+    <ItemGroup>
+      <CurrentProjectServicingConfiguration
+        Include="@(ProjectServicingConfiguration)"
+        Condition="'%(Identity)' == '$(MSBuildProjectName)'" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="PackIfServiced"
+        DependsOnTargets="
+          SkipPackIfShouldNotBuild;
+          Pack" />
+
+  <Target Name="SkipPackIfShouldNotBuild"
+        DependsOnTargets="GetCurrentProjectServicingConfiguration">
+    <PropertyGroup>
+      <MostRecentProducedServicingPatchVersion>%(CurrentProjectServicingConfiguration.PatchVersion)</MostRecentProducedServicingPatchVersion>
+      <IsPackable Condition="'$(MostRecentProducedServicingPatchVersion)' != '$(PatchVersion)'">false</IsPackable>
```

Comment:
Hmmm, I think we might actually need to skip this for source-build, to avoid pulling in prebuilts. I *think* these binaries are actually included in the SDK (not just used as references to compile against) so we need to build nupkgs every time to propagate them.

I think this would be sufficient for us to tweak later if necessary (or not use if not necessary) while avoiding a patch:

```xml
<Target Name="SkipPackIfShouldNotBuild"
        Condition="'$(AlwaysPackEvenIfNotServicing)' != 'true'"
        DependsOnTargets="GetCurrentProjectServicingConfiguration">
```

It would be nice to figure out what our end goal is and make this more unified, but it's not worth it at the moment, just need to make sure it's configurable enough. I have an overview of the complex issue source-build is hitting, here, for the future: https://github.com/dotnet/source-build/tree/release/3.1/Documentation/planning/nongranular-servicing-readiness

Reply:
Thanks - I'll add the condition.

Reply:
Looks good. Filed https://github.com/dotnet/source-build/issues/1668 to track the source-build side of this.